From 4998c53c8775f79b7e19a93ed47a0e7e65f6be03 Mon Sep 17 00:00:00 2001
From: Simon Farnsworth <simon.farnsworth@onelan.co.uk>
Date: Fri, 7 Feb 2014 12:27:50 +0000
Subject: [PATCH] vaapipostproc: Create filter surface pool if it does not
 exist

ensure_srcpad_buffer_pool tries to avoid unnecessarily deleting and
recreating filter_pool. Unfortunately, this also meant it didn't create it
if it did not exist.

Fix it to always create the buffer pool if it does not exist.

Signed-off-by: Simon Farnsworth <simon.farnsworth@onelan.co.uk>
---
 gst/vaapi/gstvaapipostproc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gst/vaapi/gstvaapipostproc.c b/gst/vaapi/gstvaapipostproc.c
index f1be89c..70cc411 100644
--- a/gst/vaapi/gstvaapipostproc.c
+++ b/gst/vaapi/gstvaapipostproc.c
@@ -1126,7 +1126,7 @@ ensure_srcpad_buffer_pool(GstVaapiPostproc *postproc, GstCaps *caps)
     gst_video_info_set_format(&vi, postproc->format,
         GST_VIDEO_INFO_WIDTH(&vi), GST_VIDEO_INFO_HEIGHT(&vi));
 
-    if (!video_info_changed(&vi, &postproc->filter_pool_info))
+    if (postproc->filter_pool && !video_info_changed(&vi, &postproc->filter_pool_info))
         return TRUE;
     postproc->filter_pool_info = vi;
 
-- 
1.8.5.3

